app/models/concerns/geocodeable.rb

```diff
@@ -4,34 +4,26 @@ module Geocodeable
   extend ActiveSupport::Concern
 
   included do
-    geocoded_by :geocode_data
+    geocoded_by :address
     after_validation :geocode, if: :should_be_geocoded?
 
+    # Skip geocoding if this flag is truthy
     attr_accessor :skip_geocoding
 
     def skip_geocoding?
-      !!skip_geocoding
+      skip_geocoding.present?
     end
 
-    # Customize by overwriting in the Geocodeable model.
-    def geocode_data
-      @geocode_data ||= address
-    end
-
-    # An array of symbols of the db columns upon which `geocode_data`
-    # depends.These are checked for changes before geocoding using
-    # `should_be_geocoded?`. By default, empty, so we don't check for changes.
-    def geocode_columns; []; end
-
-    def any_geocode_columns_changed?
-      return true if geocode_columns.blank?
-      geocode_columns.any? { |col| public_send("#{col}_changed?") }
-    end
-
-    # Overwrite in model to customize skip-geocoding logic.
+    # Should the receiving object be geocoded?
+    #
+    # By default:
+    #  - we skip geocoding if the `skip_geocoding` flag is set.
+    #  - geocode if address is blank or has changed
+    #
+    # Overwrite this method in inheriting models to customize skip-geocoding
+    # logic.
     def should_be_geocoded?
-      return false if skip_geocoding?
-      geocode_data.present? && any_geocode_columns_changed?
+      !skip_geocoding? && (address.blank? || address_changed?)
```

Comment:
Shouldn't this be `address.present?` ?

Reply:
Good catch thanks — I think this was an incomplete DeMorgan's Law application but I just ended up confusing myself. Fixed and added a test via `StolenRecord` in d30d8b7b5554d83

Reply:
Thanks for the test! Definitely makes it clear